### PR TITLE
Add public activities

### PR DESCRIPTION
**Pull request by xuanduc987**

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -45,6 +46,14 @@ def forget
     update_attribute :remember_digest, nil
   end
 
+  def feed
+    following_ids = "SELECT followed_id FROM relationships
```
Comment (dieunb) [inline app/models/user.rb]: don't use mysql query

File: app/models/activity.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class Activity < ActiveRecord::Base
+  default_scope -> {order(created_at: :desc)}
+
+  belongs_to :user
+
+  validates :user_id, presence: true
```
Comment (dieunb) [inline app/models/activity.rb]: `validates :user, presence: true`

File: app/helpers/activities_helper.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module ActivitiesHelper
+  def get_user_name user_id
+    user = User.find user_id
+    link_to current_user?(user) ? "You" : user.name, user
+  end
+
+  def get_category_name_by_lession lession_id
```
Comment (chuanhtuan) [inline app/helpers/activities_helper.rb]: remove

File: app/models/activity.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class Activity < ActiveRecord::Base
+  default_scope -> {order(created_at: :desc)}
```
Comment (chuanhtuan) [inline app/models/activity.rb]: remove all default scope

File: app/helpers/activities_helper.rb (hunk 3 above)
Comment (chuanhtuan) [inline app/helpers/activities_helper.rb]: hàm này nên trả về object category

File: app/models/activity.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+class Activity < ActiveRecord::Base
+  belongs_to :user
+
+  validates :user, presence: true
+
+  scope :recent_first,  -> {order(created_at: :desc)}
```
Comment (dieunb) [inline app/models/activity.rb]:
```
scope :recent_first, ->{order(created_at: :desc)}
```

File: app/views/static_pages/home.html.erb [hunk 6]
```diff
@@ -8,6 +8,15 @@
         <%= render 'shared/stats' %>
       </section>
     </aside>
+    <div class="col-md-8">
+      <h3>Activities Feed</h3>
+      <% if @activities.any? %>
```
Comment (dieunb) [inline app/views/static_pages/home.html.erb]: indent 
```
+      <% if @activities.any? %>
+      <ol class="activities">
+        <%= render @activities %>
+      </ol>
+      <%= will_paginate @activities %>
+    <% end %>
```

**Pull request merged**